Mathlib/Geometry/Manifold/Instances/Icc.lean

```diff
@@ -0,0 +1,201 @@
+/-
+Copyright (c) 2025 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+import Mathlib.Analysis.InnerProductSpace.Calculus
+import Mathlib.Geometry.Manifold.ContMDiff.Basic
+import Mathlib.Geometry.Manifold.Instances.Real
+import Mathlib.Geometry.Manifold.MFDeriv.FDeriv
+
+/-! # Manifold structure on real intervals
+
+The manifold structure on real intervals is defined in `Mathlib.Geometry.Manifold.Instances.Real`.
+We relate it to the manifold structure on the real line, by showing that the inclusion and
+projection are smooth, and showing that a function defined on the interval is smooth iff
+its composition with the projection is smooth on the interval in `ℝ`.
+
+We also define `1 : TangentSpace (𝓡∂ 1) z`, and relate it to `1` in the real line.
+-/
+
+open Set
+open scoped Manifold Topology
+
+noncomputable section
+
+variable
+  {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
+  {H : Type*} [TopologicalSpace H] {I : ModelWithCorners ℝ E H} {n : WithTop ℕ∞}
+  {M : Type*} [TopologicalSpace M] [ChartedSpace H M]
+
+instance (x : ℝ) : One (TangentSpace 𝓘(ℝ) x) where
+  one := (1 : ℝ)
+
+/-- Unit vector in the tangent space to a segment, as the image of the unit vector in the real line
+under the canonical projection. It is also mapped to the unit vector in the real line through
+the canonical injection, see `mfderiv_subtypeVal_Icc_one`.
+
+Note that one can not abuse defeqs for this definition: this is *not* the same as the vector
+`fun _ ↦ 1` in `EuclideanSpace ℝ (Fin 1)` through defeqs, as one of the charts of `Icc x y` is
+orientation-reversing. -/
+irreducible_def oneTangentSpaceIcc {x y : ℝ} [h : Fact (x < y)] (z : Icc x y) :
+    TangentSpace (𝓡∂ 1) z :=
+  mfderivWithin 𝓘(ℝ) (𝓡∂ 1) (Set.projIcc x y h.out.le) (Icc x y) z 1
+
+instance {x y : ℝ} [h : Fact (x < y)] (z : Icc x y) : One (TangentSpace (𝓡∂ 1) z) where
+  one := oneTangentSpaceIcc z
+
+variable {x y : ℝ} [h : Fact (x < y)] {n : WithTop ℕ∞}
+
+/-- The inclusion map from of a closed segment to `ℝ` is smooth in the manifold sense. -/
+lemma contMDiff_subtypeVal_Icc  :
+    ContMDiff (𝓡∂ 1) 𝓘(ℝ) n (fun (z : Icc x y) ↦ (z : ℝ)) := by
+  intro z
+  rw [contMDiffAt_iff]
+  refine ⟨by fun_prop, ?_⟩
+  simp? says
+    simp only [extChartAt, PartialHomeomorph.extend, PartialHomeomorph.refl_partialEquiv,
+      PartialEquiv.refl_source, PartialHomeomorph.singletonChartedSpace_chartAt_eq,
+      modelWithCornersSelf_partialEquiv, PartialEquiv.trans_refl, PartialEquiv.refl_coe,
+      Icc_chartedSpaceChartAt, PartialEquiv.coe_trans_symm, PartialHomeomorph.coe_coe_symm,
+      ModelWithCorners.toPartialEquiv_coe_symm, CompTriple.comp_eq, PartialEquiv.coe_trans,
+      ModelWithCorners.toPartialEquiv_coe, PartialHomeomorph.toFun_eq_coe, Function.comp_apply]
+  split_ifs with hz
```

Comment:
Are there any useful high-level comments you can make? If the answer is no and this proof should just be five lines if not for differential geometry painfulness, that may be fine as well.